torch/fx/experimental/Partitioner.py

```diff
@@ -106,77 +122,378 @@ def get_output_nodes(self) -> List[Node]:
+                partition.add_node(node)
+                # Add all input nodes if they are placeholders or constants
+                for n in input_nodes:
+                    if (n not in partition.nodes) and (n.op in {'placeholder', 'get_attr'}):
+                        partition.add_node(n)
+                used_mem = used_mem + total_size_of_input_nodes
+        # Update used mem mapping for the last partition
+        self.partition_to_used_mem[partition] = used_mem
+        # Find all orphans (placeholders or constants are never used)
+        # and create a partiton for each of them
         for node in self.graph_module.graph.nodes:
             if node.op == 'output':
                 break
-            node.partition_ids = self.node_to_partitions[node]
+            if len(self.users_mapping[node]) == 0 and len(self.args_mapping[node]) == 0:
+                partition = self.create_partition()
+                partition.add_node(node)
+                if hasattr(node, 'NodeSize'):
+                    self.partition_to_used_mem[partition] = node.NodeSize.total_size
+                else:
+                    raise RuntimeError('node has no NodeSize attr')
+                self.orphans.add(node)
+        # Find parent partitions and child partitions for each partition.
+        self.set_parents_and_children()
+        # Combine small partitions
+        self.combine_partitions_based_on_size(available_mem)
+        # Reassign partition ids and update self.node_to_partitions.
+        self.reorganize_partitions()
         return
 
-    def dump_partition_DAG(self) -> DAG:
+    def do_partition(self) -> GraphModule:
+        """Return a module with submodules (partitions)."""
+        for node in self.graph_module.graph.nodes:
+            if node.op == 'output':
+                break
+        module_with_submodules = split_module(
+            self.graph_module,
+            self.torch_module,
+            lambda node: self.node_to_partitions[node][0]
+        )
+        return module_with_submodules
+
+    def dump_partition_DAG(self, module_with_submodules: GraphModule) -> DAG:
         dag = DAG()
-        for i, partition in enumerate(self.partitions):
-            input_nodes = partition.get_input_nodes()
-            output_nodes = partition.get_output_nodes()
-            parents = partition.parents
-            parent_ids = [self.partitions.index(p) for p in parents]
-            children = partition.children
-            children_ids = [self.partitions.index(c) for c in children]
-            dag.create_node(i, parent_ids, children_ids, input_nodes, output_nodes)
+        users_mapping = _get_users_mapping(module_with_submodules)
+        args_mapping = _get_args_mapping(users_mapping)
+        for node in module_with_submodules.graph.nodes:
+            if node.name == 'output':
+                break
+            if node.op in {'placeholder', 'get_attr'}:
+                continue
+            if 'getitem' in node.name:
```

Comment:
Node names shouldn't be relied upon, they can be overridden by the user. Instead, this check should be:
```
import operator 
<...>
            if node.target == operator.__getitem__:
```